sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -1606,6 +1607,12 @@ public Node visitDataType(SqlBaseParser.DataTypeContext context) {
             return CollectionColumnType.array((ColumnType) visit(context.arrayTypeDefinition().dataType()));
         } else if (context.setTypeDefinition() != null) {
             return CollectionColumnType.set((ColumnType) visit(context.setTypeDefinition().dataType()));
+        } else if (context.baseDataType() != null) {
```

Comment:
I'd probably change the parser a bit, e.g.:
```
ident
    : quotedIdentifier                                       #quotedIdentifierAlternative
    | unquotedIdentifier                                   #unquotedIdentifierAlternative
    ;

quotedIdentifier
    : QUOTED_IDENTIFIER                                  #quotedIdentifier
    | BACKQUOTED_IDENTIFIER                         #backQuotedIdentifier
    ;

unquotedIdentifier
    : IDENTIFIER                                               #unquotedIdentifier
    | DIGIT_IDENTIFIER                                    #digitIdentifier
    | nonReserved                                            #unquotedIdentifier
    | COLON_IDENT                                         #colonIdentifier
    ;

...
dataType
    : unquotedIdentifier
    | baseDataType
    | objectTypeDefinition
    | arrayTypeDefinition
    | setTypeDefinition
    ;

baseDataType
    : DOUBLE PRECISION
    ;
```

and implement visitors for correspoding nodes in `dataType` instead of having the `visitDataType`, but would do it in the separate PR.